b0map.py

```diff
@@ -0,0 +1,146 @@
+
+
+def load_pixel_array(dicom_files: list[str], nechos: int = 3) -> np.ndarray:
+    """Load pixel arrays from DICOM files and reshape by echoes."""
+    echo_data = np.array(
+        [pdcm.dcmread(dcm).pixel_array for dcm in dicom_files])
+    reshaped_data = echo_data.reshape(
+        nechos, len(echo_data) // nechos, *echo_data[0].shape
+    )
+    return np.moveaxis(np.moveaxis(reshaped_data, 0, -1), 0, -2)
+
+
+def extract_echo_times(dicom_file: str) -> tuple[float, float, float]:
+    """Extract echo times (TE0, TE1, TE2) from a DICOM file."""
+    TE = []
+    with open(dicom_file, "rb") as f:
+        for line in f.readlines():
+            if line[:7] in [b"alTE[0]", b"alTE[1]", b"alTE[2]"]:
+                TE.append(int(line.decode("ascii").split()[-1]))
+    if len(TE) != 3:
+        raise ValueError(
+            "Could not find all three echo times in DICOM header.")
+    return TE[0], TE[1], TE[2]
+
+
+def load_magnitude_from_dicom(folder_path: str) -> np.ndarray:
+    """Load magnitude data from DICOM files."""
+    dicom_files = load_dicom_files(folder_path)
+    return load_pixel_array(dicom_files)
+
+
+def create_mask(magnitude: np.ndarray, mask_type: str) -> np.ndarray:
+    """Create a mask based on the specified method."""
+    if mask_type == "Matlab":
+        mat_data = scipy.io.loadmat(
+            '/volatile/home/st281428/field_map/B0/_1/Mask.mat')
+        return mat_data['Mask']
+    elif mask_type == "Python":
+        mean_amplitude = np.mean(magnitude, axis=-1)
+        smoothed_amplitude = gaussian_filter(mean_amplitude, sigma=1)
+        threshold = np.percentile(mean_amplitude, 62)
+        return smoothed_amplitude > threshold
+    elif mask_type == "Phantom":
+        mask = gpm.get_phantom_mask_from_snr(np.sum(magnitude, axis=-1))
+        return ndimage.binary_fill_holes(mask)
+    else:
+        raise ValueError(f"Unknown mask type: {mask_type}")
+
+
+def unwrap_phase_3d(field_map: np.ndarray, dte1: float, dte2: float) -> np.ndarray:
+    """Perform 3D phase unwrapping and calculate B0 map."""
+    tau = np.array([0.0, dte1, dte1 + dte2])
+    sum_tau, sum_tau2 = np.sum(tau), np.sum(tau * tau)
+
+    diff = field_map[:, :, :, 1] - field_map[:, :, :, 0]
+    dunwrapped = sr.unwrap_phase(diff) / (2 * np.pi)
+
+    D = [field_map[:, :, :, 0] / (2 * np.pi), None, None]
+    D[1] = D[0] + dunwrapped
+    D[2] = field_map[:, :, :, 2] / (2 * np.pi)
+
+    d2_theoretical = dunwrapped * (dte1 + dte2) / dte1
+    D[2] += np.round(d2_theoretical)
+    eps = (D[2] - D[0]) - d2_theoretical
+    D2neg = (eps < -0.5)
+    D2pos = (eps >= 0.5)
+    D[2] += D2neg.astype(np.int32) - D2pos.astype(np.int32)
+
+    sxy = tau[1] * D[1] + tau[2] * D[2]
+    sy = D[0] + D[1] + D[2]
+    return np.round((3 * sxy - sum_tau * sy) / (3 * sum_tau2 - sum_tau ** 2))
+
+
+def load_field_map_from_dicom(
+    folder_path: str, nx: int, ny: int, nz: int, unwrap: str = "3D", mask_type: str = "Phantom"
```

Comment:
Why is unwrap default called "3D"? 
What you mean is triple echo? Ideally unwrap happens always, the question is how many echoes do you want to use for estimating B0 map. Perhaps add it as an arg called : `num_echoes_for_estimate` or `num_echoes` 
Again, please add a docstring here.